File: src/com/developerphil/adbidea/ui/NotificationHelper.java
package com.developerphil.adbidea.ui;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import com.intellij.notification.Notifications;

/**
 * Created by pbreault on 1/2/14.
 */
public class NotificationHelper {
    public static void info(String message) {
        sendNotification(message, NotificationType.INFORMATION);
    }

    public static void error(String message) {
        sendNotification(message, NotificationType.ERROR);
    }

    public static void sendNotification(String message, NotificationType notificationType) {
        Notification notification = new Notification("com.developerphil.adbidea", "ADB IDEA", espaceString(message), notificationType);
        Notifications.Bus.notify(notification);
    }


    private static String espaceString(String string) {
        // replace with both so that it returns are preserved in the notification ballon and in the event log
        return string.replaceAll("\n", "\n<br />");
    }
}

File: src/main/kotlin/com/developerphil/adbidea/ui/NotificationHelper.kt
package com.developerphil.adbidea.ui

import com.intellij.notification.NotificationGroup
import com.intellij.notification.NotificationGroupManager
import com.intellij.notification.NotificationType

object NotificationHelper {
    fun info(message: String) {
        sendNotification(
            message,
            NotificationType.INFORMATION,
            NotificationGroupManager
                .getInstance()
                .getNotificationGroup("ADB Idea (Logging)")
        )
    }

    // Function to send an error notification
    fun error(message: String) {
        sendNotification(
            message,
            NotificationType.ERROR,
            NotificationGroupManager
                .getInstance()
                .getNotificationGroup("ADB Idea (Errors)")
        )
    }

    // Helper function to create and display a notification
    private fun sendNotification(
        message: String,
        notificationType: NotificationType,
        notificationGroup: NotificationGroup,
    ) {
        // Create the notification without a listener
        val notification = notificationGroup.createNotification(
            "ADB IDEA",
            escapeString(message),
            notificationType,
        )

        // Display the notification
        notification.notify(null)
    }

    private fun escapeString(string: String) = string.replace(
        "\n".toRegex(),
        "\n<br />"
    )
}